Docs/Build-DSC.md

```diff
@@ -0,0 +1,60 @@
+## Table of Contents
+- [Setup DSC and OMI Repo](#setup-dsc-and-omi-repo)
+- [Build OMI](#build-omi)
+- [Build DSC](#build-dsc)
+- [Run OMI Server](#run-omi-server)
+- [Test DSC basic functionality](#test-dsc-basic-functionality)
+- [Update DSC and rebuild](#update-dsc-and-rebuild)
+
+
+## Setup DSC and OMI Repo
+
+1. Clone the repo. 
+   - git clone https://github.com/Microsoft/PowerShell-DSC-for-Linux.git 
+   - cd PowerShell-DSC-for-Linux
+2. mkdir omi-1.3.0-2 
+3. mkdir omi-1.0.8 
+4. Download the source code for this release - https://github.com/Microsoft/omi/releases/tag/v1.3.0-2 
+5. Unzip it under omi-1.3.0-2
+6. Copy over everything from under omi-1.3.0-2/Unix to omi-1.0.8
+7. Set up the dependent packages for OMI from https://github.com/Microsoft/Build-omi 
+   - On Ubuntu 14.04 and 16.04
+        - sudo apt-get install git pkg-config make g++ rpm librpm-dev libpam0g-dev libssl-dev libkrb5-dev gawk
+8. Set up dependent packages for DSC 
+   - sudo apt-get install libcurl4-gnutls-dev
+
+## Build OMI
+
+1. cd omi-1.3.0-2/Unix
+2. ./configure -–dev –-enable-debug
+3. make -j
+4. Copy everything from omi-1.3.0-2/Unix/Output to omi-1.0.8/ (This is a temporary workaround and needs to be fixed)
+
+## Build DSC
+
+1. cd PowerShell-DSC-for-Linux
+2. ./configure --no-rpm --no-dpkg –local –-enable-debug
+3. make -j
+4. Open LCM/GNUmakefile. There will be 3 occurrences of the word ‘ConsistencyInvoker’. comment out the last 2 occurrences that mention ‘ConsistencyInvoker’
+5. make reg
+
+## Run OMI Server
+
+1. cd omi-1.3.0-2/Unix/Output/bin
+2. sudo omiserver -d
+
+## Test DSC basic functionality
+
+1. cd PowerShell-DSC-for-Linux/intermediate/Scripts
+2. sudo ./GetDscLocalConfigurationManager.py
+
+## Update DSC and rebuild 
+As part of developer workflow, anytime we make changes to DSC code, these are the steps we need to follow
+1. Sudo killall omiserver           # Kill OMI Server before building OMI
```

Comment:
what building omi or dsc has got to do with killing omiserver. This seems something we should not do at all.

Reply:
This is because we are deleting output folders from omi folder for rebuilding DSC The omiserver is loaded from omi-1.3.0-2/Unix/output/bin directory.

Reply:
There is no need to do that. It might somehow work by following these instructions. But its very inefficient and time consuming workflow which we don't have to follow.

Reply:
In case I do not delete the output folders, I get the following error 
/home/insivara/Git/DSC-Linux/omi-1.3.0-2/Unix/output/lib/libmofparser.a(ptrarray.o): In function `PtrArray_Append':
/home/insivara/Git/DSC-Linux/LCM/codec/mof/parser/ptrarray.c:90: multiple definition of `PtrArray_Append'
/home/insivara/Git/DSC-Linux/omi-1.3.0-2/Unix/output/lib/libbase.a(ptrarray.o):/home/insivara/Git/DSC-Linux/omi-1.3.0-2/Unix/base/ptrarray.c:62: first defined here
collect2: error: ld returned 1 exit status
make[5]: *** [/home/insivara/Git/DSC-Linux/omi-1.3.0-2/Unix/output/lib/libtest_cli.so] Error 1
make[5]: Leaving directory `/home/insivara/Git/DSC-Linux/omi-1.3.0-2/Unix/tests/cli'
make[4]: *** [all] Error 2
make[4]: Leaving directory `/home/insivara/Git/DSC-Linux/omi-1.3.0-2/Unix'
make[3]: *** [all] Error 2
make[3]: Leaving directory `/home/insivara/Git/DSC-Linux/omi-1.0.8'
make[2]: *** [lcm] Error 2
make[2]: Leaving directory `/home/insivara/Git/DSC-Linux'
make[1]: *** [local] Error 2
make[1]: Leaving directory `/home/insivara/Git/DSC-Linux'
make: *** [all] Error 2